sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/AttributeSet.scala

```diff
@@ -39,10 +41,15 @@ object AttributeSet {
 
   /** Constructs a new [[AttributeSet]] given a sequence of [[Expression Expressions]]. */
   def apply(baseSet: Iterable[Expression]): AttributeSet = {
-    new AttributeSet(
-      baseSet
-        .flatMap(_.references)
-        .map(new AttributeEquals(_)).toSet)
```

Comment:
I think this is just duplicating the code in `fromAttributeSets`, right? What would be the benefit of doing that?

Reply:
The simple one is better? https://github.com/maropu/spark/commit/cba767e3721e78b934ef02631378435eb3aef730

Reply:
ok, so there are 2 reasons why the current one is better than the proposed:

 - in your proposal we are calling `_.references` also in cases when it is not needed. This causes unneeded operations (basically https://github.com/apache/spark/pull/22364/files#diff-75576f0ec7f9d8b5032000245217d233R40 is called for each attribute... see also `Attribute.references`);
 - the current one uses a mutable Set, but this is not a big deal, it can be changed.

Reply:
ah, ok. can you rename `fromAttributeSets` -> `apply`, and then change the mutable set?

Reply:
I cannot rename to `apply` because there is already an `apply` with `Iterable` and we cannot have 2 `apply` one with `Iterable[Expression]` and one with `Iterable[AbstractSet]` because they would be conflicting.

Reply:
The name is ok. We still need to use the mutable set instead of `new AttributeSet(sets.foldLeft(Set.empty[AttributeEquals])(_ ++ _.baseSet))`?

Reply:
I have to use a `LinkedHashSet` because without keeping the order of insertion I had some UT failures. But I can use the more compact syntax you suggested here, thanks.

Reply:
Thanks!

Reply:
Thank you for your comment and help here!